include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -525,11 +452,159 @@ class interleaved_bloom_filter
         archive(bin_words);
         archive(hash_funs);
         archive(data);
-        result_buffer.resize(bins);
     }
     //!\endcond
 };
 
+/*!\brief Manages queries for the seqan3::interleaved_bloom_filter.
+ * \attention Calling seqan3::interleaved_bloom_filter::increase_bin_number_to on `ibf` invalidates the
+ * membership_agent.
+ *
+ * \details
+ *
+ * ### Example
+ *
+ * \include test/snippet/search/dream_index/membership_agent_construction.cpp
+ */
+template <data_layout data_layout_mode>
+class interleaved_bloom_filter<data_layout_mode>::membership_agent
+{
+private:
+    //!\brief The type of the augmented seqan3::interleaved_bloom_filter.
+    using ibf_t = interleaved_bloom_filter<data_layout_mode>;
+
+    //!\brief A pointer to the augmented seqan3::interleaved_bloom_filter.
+    ibf_t const * ibf_ptr{nullptr};
+
+public:
+    class binning_bitvector;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    membership_agent() = default; //!< Defaulted.
+    membership_agent(membership_agent const &) = default; //!< Defaulted.
+    membership_agent & operator=(membership_agent const &) = default; //!< Defaulted.
+    membership_agent(membership_agent &&) = default; //!< Defaulted.
+    membership_agent & operator=(membership_agent &&) = default; //!< Defaulted.
+    ~membership_agent() = default; //!< Defaulted.
+
+    /*!\brief Construct a membership_agent for an existing seqan3::interleaved_bloom_filter.
+     * \private
```

Comment:
?
Why only make the param private? (if that was the intention at all)

Reply:
This marks the constructor as private, not just the param.